File: TRPO/lab6/project-root/src/main/kotlin/UserType.kt
import java.io.InputStream

interface Comparator {
    fun compare(o1: Any?, o2: Any?): Int
}

interface UserType {
    fun typeName(): String
    fun create(): Any
    fun clone(obj: Any): Any
    fun readValue(input: InputStream): Any
    fun parseValue(ss: String): Any
    fun getComparator(): Comparator
}

File: TRPO/lab5/scala_lab5/src/main/scala/UserType.scala
trait Comparator {
  def compare(o1: Any, o2: Any): Int
}

trait UserType {
  def typeName: String
  def create: Any
  def clone(obj: Any): Any
  def readValue(in: java.io.InputStreamReader): Any
  def parseValue(ss: String): Any
  def getTypeComparator: Comparator
}